ansible_collections/arista/avd/tests/unit/filters/test_ipaddr.py

```diff
@@ -0,0 +1,151 @@
+    def test_ipaddr_subnetwork_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='subnet')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'subnet') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_NETMASK_EXPECTED)])
+    def test_ipaddr_netmask_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='netmask')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'netmask') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_PREFIX_EXPECTED)])
+    def test_ipaddr_prefixes_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='prefix')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'prefix') == IP_ADDRESS_LIST_EXPECTED
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_SIZE_EXPECTED)])
+    def test_ipaddr_size_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='size')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'size') == IP_ADDRESS_LIST_EXPECTED
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_NETWORK_ADDRESSES_EXPECTED)])
+    def test_ipaddr_network_address_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='network')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'network') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("IP_ADDRESS_LIST_INPUT,IP_ADDRESS_LIST_EXPECTED", [(IP_ADDRESSES_INPUT, IP_ADDRESSES_EXPECTED)])
+    def test_ipaddr_network_address_list(self, IP_ADDRESS_LIST_INPUT, IP_ADDRESS_LIST_EXPECTED):
+        resp_list = f.ipaddr(IP_ADDRESS_LIST_INPUT, method='address')
+        assert resp_list == IP_ADDRESS_LIST_EXPECTED
+        assert get_by_method(IP_ADDRESS_LIST_INPUT, 'address') == IP_ADDRESS_LIST_EXPECTED
+        ipv4_ipv6_assert(resp_list, IP_ADDRESS_LIST_EXPECTED)
+
+    @pytest.mark.parametrize("INPUT_LISTS,IP_ADDRESS_LISTS_EXPECTED", [((IP_ADDRESSES_INPUT, DEFAULTATTR_INPUT), IP_ADDRESSES_DEFAULTATTR_EXPECTED)])
```

Comment:
you should really add a third input parameter instead of overloading on of them like this.